cmd/vega/node/node.go

```diff
@@ -284,8 +284,14 @@ func (n *NodeCommand) startAPIs() error {
 
 	n.proxyServer = rest.NewProxyServer(n.Log, n.conf.API)
 
-	if bool(n.conf.Admin.Server.Enabled) && n.conf.IsValidator() {
-		adminServer, err := admin.NewServer(n.Log, n.conf.Admin, n.vegaPaths, n.nodeWalletPassphrase, n.nodeWallets)
+	if n.conf.IsValidator() {
+		adminServer, err := admin.NewValidatorServer(n.Log, n.conf.Admin, n.vegaPaths, n.nodeWalletPassphrase, n.nodeWallets, n.protocol.GetProtocolUpgradeService())
+		if err != nil {
+			return err
+		}
+		n.adminServer = adminServer
+	} else {
+		adminServer, err := admin.NewNonValidatorServer(n.Log, n.conf.Admin, n.vegaPaths, n.protocol.GetProtocolUpgradeService())
```

Comment:
Alright you do have 2 constructors to instantiate it in a validator or non-validators mode indeed.